kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,575 @@
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
+            hasNoSuchElementExceptionThrown = true;
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+            hasNullPointerExceptionThrown = true;
+        }
+    }
+    
+    public void whenGetInterfaceStatus(String netInterface, NetworkService netService) {
+        try {
+            this.netInterface = this.instanceNMDbusConnector.getInterfaceStatus(netInterface, netService);
+        } catch (DBusException e) {
+            e.printStackTrace();
+            hasDBusExceptionBeenThrown = true;
+        } catch (NoSuchElementException e) {
+            e.printStackTrace();
+            hasNoSuchElementExceptionThrown = true;
+        } catch (NullPointerException e) {
+            e.printStackTrace();
+            hasNullPointerExceptionThrown = true;
+        }
+    }
+
+    public void thenNoExceptionIsThrown() {
+        assertFalse(hasDBusExceptionBeenThrown);
+        assertFalse(hasNoSuchElementExceptionThrown);
+    }
+
+    public void thenDBusExceptionIsThrown() {
+        assertTrue(hasDBusExceptionBeenThrown);
+    }
+
+    public void thenNullPointerExceptionIsThrown() {
+        assertTrue(hasNullPointerExceptionThrown);
+    }
+
+    public void thenNoSuchElementExceptionIsThrown() {
+        assertTrue(hasNoSuchElementExceptionThrown);
+    }
+
+    public void thenGetDbusConnectionIsMockedConnection() {
+        assertEquals(this.dbusConnection, this.dbusConnectionInternal);
+    }
+
+    public void thenConnectionClosed() {
+        verify(this.dbusConnection, atLeastOnce()).disconnect();
+    }
+
+    public void thenCheckVersionIsRun() throws DBusException {
+        verify(this.dbusConnection, atLeastOnce()).getRemoteObject(eq("org.freedesktop.NetworkManager"),
+                eq("/org/freedesktop/NetworkManager"), any());
+    }
+
+    public void thenCheckPermissionsRan() {
+        verify(this.mockedNetworkManager, atLeastOnce()).GetPermissions();
+    }
+
+    public void thenReturnedDevicesAre(List<String> list) {
+        assertEquals(list, this.internalStringList);
+    }
+    
+    public void thenDisconnectIsCalledFor(String netInterface) {
+        verify(this.mockDevices.get(netInterface)).Disconnect();
+    }
+    
+    public void thenConnectDisconnectIsCalledFor(String netInterface) throws DBusException {
+        Connection connect = this.dbusConnection.getRemoteObject("org.freedesktop.NetworkManager", "/mock/device/" + netInterface, Connection.class);
+        verify(connect).Update(any());
+    }
+    
+    public void thenNetoworkSettingsDoNotChangeForDevice(String netInterface) throws DBusException {
+        
+        Connection connect = this.dbusConnection.getRemoteObject("org.freedesktop.NetworkManager", "/mock/device/" + netInterface, Connection.class);
+        verify(connect, never()).Update(any());
+        verify(this.mockDevices.get(netInterface), never()).Disconnect();
```

Comment:
You should also check that the `NetworkManager.ActivateConnection` and `NetworkManager.AddAndActivateConnection` are not called.